samples/wifi/radio_test/src/nrf_wifi_radio_ficr_shell.c

```diff
@@ -0,0 +1,345 @@
+
+	if (flags & (~MAC0_ADDR_FLAG_MASK))
+		LOG_INF("MAC0 Address are not programmed in OTP\n");
+	else
+		LOG_INF("MAC0 Address is programmed in OTP\n");
+
+	if (flags & (~MAC1_ADDR_FLAG_MASK))
+		LOG_INF("MAC1 Address are not programmed in OTP\n");
+	else
+		LOG_INF("MAC1 Address is programmed in OTP\n");
+
+	if (flags & (~CALIB_XO_FLAG_MASK))
+		LOG_INF("CALIB_XO is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_XO is programmed in OTP\n");
+
+	if (flags & (~CALIB_PDADJMCS7_FLAG_MASK))
+		LOG_INF("CALIB_PDADJMCS7 not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_PDADJMCS7 is programmed in OTP\n");
+
+	if (flags & (~CALIB_PDADJMCS0_FLAG_MASK))
+		LOG_INF("CALIB_PDADJMCS0 is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_PDADJMCS0 is programmed in OTP\n");
+
+	if (flags & (~CALIB_MAXPOW2G4_FLAG_MASK))
+		LOG_INF("CALIB_MAXPOW2G4 is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_MAXPOW2G4 is programmed in OTP\n");
+
+	if (flags & (~CALIB_MAXPOW5G0MCS7_FLAG_MASK))
+		LOG_INF("CALIB_MAXPOW5G0MCS7 is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_MAXPOW5G0MCS7 is programmed in OTP\n");
+
+	if (flags & (~CALIB_MAXPOW5G0MCS0_FLAG_MASK))
+		LOG_INF("CALIB_MAXPOW5G0MCS0 is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_MAXPOW5G0MCS0 is programmed in OTP\n");
+
+	if (flags & (~CALIB_RXGAINOFFSET_FLAG_MASK))
+		LOG_INF("CALIB_RXGAINOFFSET is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_RXGAINOFFSET is programmed in OTP\n");
+
+	if (flags & (~CALIB_TXPOWBACKOFFT_FLAG_MASK))
+		LOG_INF("CALIB_TXPOWBACKOFFT is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_TXPOWBACKOFFT is programmed in OTP\n");
+
+	if (flags & (~CALIB_TXPOWBACKOFFV_FLAG_MASK))
+		LOG_INF("CALIB_TXPOWBACKOFFV is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_TXPOWBACKOFFV is programmed in OTP\n");
+}
+
+
+static int nrf_wifi_radio_test_otp_get_status(const struct shell *shell,
+					size_t argc,
+					char  *argv[])
+{
+	unsigned int ret, err;
+	unsigned int val[OTP_MAX_WORD_LEN];
+
+	/* read all the OTP memory */
+	err = read_otp_memory(0, &val[0], OTP_MAX_WORD_LEN);
+	if (err) {
+		LOG_ERR("FAILED reading otp memory......\n");
+		return -ENOEXEC;
+	}
+
+	LOG_INF("Checking OTP PROTECT Region......\n");
+	ret = check_protection(&val[0], REGION_PROTECT, REGION_PROTECT+1,
+							REGION_PROTECT+2, REGION_PROTECT+3);
+
+	disp_location_status("OTP", ret);
+	disp_fields_status(val[REGION_DEFAULTS]);
+
+	return 0;
+}
+
+static int nrf_wifi_radio_test_otp_read_params(const struct shell *shell,
+					size_t argc,
+					char  *argv[])
+{
+	unsigned int val[OTP_MAX_WORD_LEN];
+	unsigned int ret, err;
+
+
+	err = read_otp_memory(0, &val[0], OTP_MAX_WORD_LEN);
+	if (err) {
+		LOG_ERR("FAILED reading otp memory......\n");
+		return -ENOEXEC;
+	}
+
+	ret = check_protection(&val[0], REGION_PROTECT+0, REGION_PROTECT+1,
+							REGION_PROTECT+2, REGION_PROTECT+3);
+
+	disp_location_status("OTP", ret);
+
+	LOG_INF("REGION_PROTECT0 = 0x%08x\n", val[REGION_PROTECT+0]);
```

Comment:
A general comment regarding logger usage in this module. It seems that it is used to print repsponse for the shell command. It could be the case, that logger and shell are configured to use different mediums (for example shell over UART / logger over RTT). In such case user won't see the command output.

A better choice here would be to use `shell_fprintf()` function, to make sure the command output is printed over the same medium as shell.